### PR TITLE
player_api: set_texture() to set single textures

### PR DESCRIPTION
**Pull request by Zughy**
I saw the player_api redo PR so I thought about doing something too. Right now in mods such as 3D armor I can't change just one texture (for example, to update the item I'm currently wielding). It updates every texture: wielded item, skin, armor. I thought about adding a function where one and one texture only is updated

### CODE REVIEW

**Comment by paramat** [comment]
> Right now in mods such as 3D armor I can't change just one texture (for example, to update the item I'm currently wielding). It updates every texture: wielded item, skin, armor.

But your proposed feature does this too, it gets all textures then sets all textures with only one texture changed.

MTG is feature frozen so how is this essential or extremely useful?
What do mods have to do currently that is a problem?
Is this a convenience function to avoid stating all textures when changing just one?

The other work on player_api is being done as a bugfix and to complete a feature that was in progress before feature freeze. New features are not being accepted into player_api unless essential.

**Comment by Zughy** [comment]
> What do mods have to do currently that is a problem?

Is it fair if I say "mine"? I have a mod which colorise the player texture, but as soon as I change the item in my hand (I have 3d armor), the color is gone as 3D armor replaces everything. At the same time, on another "wardrobe-like" mod of mine, when a player equips a skin I can't use player_api:set_textures() because, if there is something more such as 3D armor, it'll delete the other entries. 

In fact, [this](https://github.com/stujones11/minetest-3d_armor/blob/master/3d_armor/init.lua#L240) is how 3d_armor currently manages textures, declaring a first `skin` entry to fill the gap. It uses a [list of exceptions](https://github.com/stujones11/minetest-3d_armor/blob/master/3d_armor/api.lua#L396) to avoid conflicts with other mods changing the player texture as `skins` and `wardrobe`. But if there were 100 mods changing skins, should it support 100 exceptions? And should the creator of each one of them (in this case me) wait for an eventual 3d_armor creator's approval?

> Is this a convenience function to avoid stating all textures when changing just one?

Yes

**Comment by paramat** [comment]
I think i understand now, thanks.
This is not a low-priority convenience function, it is very useful. So i think it qualifies to be considered.

**Comment by sfan5** [comment]
Missing documentation in `game_api.txt`.

**Comment by Zughy** [comment]
@sfan5 done

File: game_api.txt
Comment (sfan5) [inline game_api.txt:469]:
```suggestion
	* `index`: the index of the texture to change. If nil, default is `1`
```
Please rename it in the code too.

File: mods/player_api/api.lua
Comment (sfan5) [inline mods/player_api/api.lua:75]:
```suggestion
	texture_ID = texture_ID or 1
```

**Comment by Zughy** [comment]
@sfan5 done

**Review by SmallJoker — APPROVED** [review]
Okay sure, why not :shrug: 
LGTM

**Comment by sfan5** [comment]
```
19:33 <+MTDiscord> <luatic> sfan5: the given implementation looks incorrect to me
19:34 <+MTDiscord> <luatic> it doesn't update the internal player_textures table
19:34 <+MTDiscord> <luatic> so it creates an inconsistency: player properties are changed, but player_api.get_animation will still happily return the outdated textures
```

**Comment by Zughy** [comment]
@sfan5 done. @appgurueu thanks for pointing that out

**Comment by appgurueu** [comment]
@Zughy I don't think `player_api.get_textures` is a thing...

**Comment by Zughy** [comment]
sorry, thought about copying, it seemed reciprocal. So, I don't really care about this PR anymore (it's more than one year old) and I see appgurueu implementing their way around. Hence the question: should I really resolve this or are core devs going with the whole rewrite anyway?

**Comment by Zughy** [comment]
I've just noticed @appgurueu PR is scheduled for 5.5.0 as well. Since they implemented this function in that PR and I'm not interested in this moment to test and fix this, just go for that one (also because merging both will cause some rebase I think). I'd appreciate, though, to be added as a co-author of that specific commit, as they basically took this PR and changed a few parts. Closing

**Pull request closed without merging**
